pretty_midi/pretty_midi.py

```diff
@@ -117,6 +146,38 @@ def _load_tempo_changes(self, midi_data):
                     if tick_scale != last_tick_scale:
                         self.__tick_scales.append((event.tick, tick_scale))
 
+    def _load_metadata(self, midi_data):
+        """Populates the lists of tempo key and time signature changes
+
+        Populates self.__time_signatures with TimeSignature objects.
+            It can be accessed by the get_time_signatures method.
+        Populates self.__key_changes with KeySignature objects.
+            It can be accessed by the get_key_signatures method.
+
+        Parameters
+        ----------
+            - midi_data : midi.FileReader
+                MIDI object from which data will be read
+        """
+
+        # list to store key signature changes
+        self.__key_changes = []
```

Comment:
Let's do as you say, which is to transfer responsibility to the users, let them shoot themselves in the foot and make it public.